src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -53,53 +62,55 @@ public void testDoGet() throws IOException, ServletException {
   }
 
   @Test
-  public void testDoPost_BadUsername() throws IOException, ServletException {
-    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+  public void testDoPost_IncorrectPassword() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("wrong password");
+
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
+    loginServlet.setUserStore(mockUserStore);
+    Mockito.when(mockUserStore.getUser("test username")).thenReturn(user);
+
+    HttpSession mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
 
     loginServlet.doPost(mockRequest, mockResponse);
 
-    Mockito.verify(mockRequest)
-        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
-    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
```

Comment:
Why don't you have the old line 63 after new line 79 and 96 while having both line 115 and 116? 
``` 
Mockito.verify(mockSession).setAttribute("user", "test username");
Mockito.verify(mockResponse).sendRedirect("/conversations");